### ISSUE TITLE
Check if we can make flags case insensitive (e.g. --baseUrl)

### ISSUE THREAD

**Issue by frjo** [gohugoio/hugo#11066]
This issue https://github.com/gohugoio/hugo/issues/10949 got marked as fixed but I still see this problem unfortunately.

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.112.7+extended darwin/amd64 BuildDate=unknown
hugo v0.113.0-DEV-536bf71abe861a74354a62eb880529a44434a4fa+extended darwin/amd64 BuildDate=2023-06-02T07:19:22Z
</pre>

### Does this issue reproduce with the latest release?

Yes

**Comment by jmooring** [comment]
The documented flag is `--baseURL` (not `--baseUrl`). 

In v0.111.3 and earlier it looks like you could use either one, but as far as I can tell that was undocumented behavior. 

Hugo's CLI flags have always been case sensitive.

**Comment by bep** [comment]
What I think happened was this:

* In Hugo < 0.112.x we did no validation of sub commands because of a bug/known issue in Cobra (the library used for CLI).
* This was unfortunate for several reasons (mostly as you lost the "did you mean?"
* That meant that some flag values was just passed as-is into Hugo's config, which is "case insensitive", and that made it work by accident.
* I fixed the sub command thing in Hugo 0.112, which means we now get some unexpected (but correct) errors.

I don't mind having case insensitive flags if someone can find a setting for it.
